registry/storage/catalog.go

```diff
@@ -49,17 +48,22 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
```

Comment:
Currently, if the array size check will be fulfilled but it is also the last repo, we want to return `io.EOF` instead of `errFinishedWalk` that will end up being dropped. So this behaviour makes sense but I also had that same question when I saw it.